### PR TITLE
TW 382 Unresolved variables and general refresh of variables page

### PR DESCRIPTION
**Pull request by HeatherStenson**
* Refresh for /docs/sending-requests/variables/
* New section about unresolved variables

### CODE REVIEW

File: src/pages/docs/sending-requests/variables.md [hunk 1]
```diff
@@ -331,55 +361,91 @@ pm.environment.get("variable_key");
...
-## Logging variables
+For more information on working with variables as a team, see [Managing environments](/docs/sending-requests/managing-environments/#creating-environments).
 
-You can log variable values to the [Postman Console](/docs/sending-requests/troubleshooting-api-requests/) while your requests run. Open the console from the button on the bottom left of Postman, or from the __View__ menu. To log the value of a variable, use the following syntax in your script:
+> Local and data variables have current values, which don't persist beyond request or collection runs.
 
-```js
-console.log(pm.variables.get("variable_key"));
-```
+## Fixing unresolved variables
```
Comment (HeatherStenson) [inline src/pages/docs/sending-requests/variables.md:414]: @malvikach This is the new section about unresolved variables. There are two other mentions that I'd like to have you review as well - I will @ you in those sections. The rest of the changes in this file are a general page refresh and you don't need to review them (unless you want to 😁 ).

File: src/pages/docs/sending-requests/troubleshooting-api-requests.md [hunk 2]
```diff
@@ -108,6 +108,7 @@ Issue | Resolving the issue
 **Invalid responses** | If your server sends incorrect response encoding errors, or invalid headers, Postman may fail to interpret the response.
 **TLS version** | Postman supports TLS version 1.2 and higher, which [may not be supported if you are using an older browser or operating system](https://support.postman.com/hc/en-us/articles/360041392573-Deprecating-TLS-1-0-and-TLS-1-1).
 **Postman errors** | It is possible that Postman might be making invalid requests to your API server. You can confirm this by checking your server logs, if available. If you believe this is happening, contact the Postman team using the [GitHub issue tracker](https://github.com/postmanlabs/postman-app-support/issues).
+**Unresolved variables** | An unresolved variable is not defined in an active scope that is available for the request it’s used in. For more information on why this happens and how to solve the problem, see [Fixing unresolved variables](/docs/sending-requests/variables/#fixing-unresolved-variables).
```
Comment (HeatherStenson) [inline src/pages/docs/sending-requests/troubleshooting-api-requests.md:111]: @malvikach This new entry in the common errors page links back to the new section about unresolved variables.

File: src/pages/docs/sending-requests/variables.md [hunk 3]
```diff
@@ -302,17 +320,29 @@ You can use variables in request URLs, parameters, headers, authorization, body,
...
 
 ![Variable Prompt](https://assets.postman.com/postman-docs/var-prompt.jpg)
 
-The prompt will indicate current value, scope (highlighted by color), and overridden status where relevant.
+The prompt indicates the current value, scope (highlighted by color), and overridden status where relevant.
 
 ![Overridden Variable](https://assets.postman.com/postman-docs/overridden-var.jpg)
 
-If a variable is unresolved, Postman will highlight it in red.
+If a variable is unresolved, Postman highlights it in red. For information on how to fix an unresolved variable, see [Fixing unresolved variables](#fixing-unresolved-variables).
```
Comment (HeatherStenson) [inline src/pages/docs/sending-requests/variables.md:331]: @malvikach I left this mention here and added a link to the new section about unresolved variables.

**Comment by HeatherStenson** [comment]
This still needs updated screenshots for unresolved variables, but I'm removing `draft` status to make this easier to review.

File: src/pages/docs/sending-requests/variables.md [hunk 4]
```diff
@@ -50,7 +50,7 @@ warning: false
 
 ---
 
-Variables enable you to store and reuse values in your requests and scripts. By storing a value in a variable, you can reference it throughout your collections, environments, and requests. If you need to update the value, you only have to change it in one place. Using variables increases your ability to work efficiently and minimizes the likelihood of error.
+_Variables_ enable you to store and reuse values in Postman. By storing a value as a variable, you can reference it throughout your collections, environments, requests, and test scripts. If you need to update the value of a variable, you can make one change and the new value is available for you wherever you use the variable. Variables help you work efficiently, collaborate with teammates, and set up dynamic workflows.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:53]: "If you need to update the value of a variable, you can make one change and the new value is available for you wherever you use the variable." — I'm not sure this sentence is necessary, maybe consider rephrasing or removing it

File: src/pages/docs/sending-requests/variables.md [hunk 5]
```diff
@@ -59,121 +59,114 @@ Variables enable you to store and reuse values in your requests and scripts. By
...
-1. Open a new request tab and enter `https://postman-echo.com/get?var={{my_variable}}` as the URL. Hover over the variable name and you'll see the value.
+1. Select <img alt="Save icon" src="https://assets.postman.com/postman-docs/icon-save.jpg" width="16px" style="vertical-align:middle;margin-bottom:5px"> **Save** and close the environment dialog.
+1. Open a new request tab and enter `https://postman-echo.com/get?var={{my_variable}}` as the URL. Hover over the variable name and you'll see the variable's value and scope.
+
+    <img alt="View the variable value and scope" src="https://assets.postman.com/postman-docs/variable-quick-start-hover.jpg" width="525px"/>
+
 1. Select __Send__ and send the request. In the response, you'll see that Postman sent the variable value to the API.
 1. Try changing the value in the Environment quick look and sending the request again.
 
 Read on for more detail on how you can use variables in Postman.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:82]: This is a bit awkward, maybe can remove or reword?

File: src/pages/docs/sending-requests/variables.md [hunk 6]
```diff
@@ -59,121 +59,114 @@ Variables enable you to store and reuse values in your requests and scripts. By
...
-
-> Postman will store environment and global variables as strings. If you’re storing objects or arrays, remember to `JSON.stringify()` them before storing, and `JSON.parse()` them when you retrieve them.
-
 Variables in Postman are key-value pairs. Each variable name represents its key, so referencing the variable name allows you to access its value.
 
-For example, if you have a base URL for requests stored in a variable named `base_url`, you can reference it in your requests using `{{base_url}}`. Whatever value is stored in the variable will be included wherever you've referenced the variable when your requests run. If the base URL value is `https://httpbin.org`, and is listed as part of the request URL using `{{base_url}}/get?customers=new`, Postman will send the request to `https://httpbin.org/get?customers=new`.
+You can use variables to pass data between requests and tests, for example if you are [chaining requests](/docs/running-collections/intro-to-collection-runs/) using a collection.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:98]: I don't think we explicitly go over chaining requests in the linked doc, or really anywhere (though we should..), so for the purpose of this doc it might be more helpful to direct users to this public workspace.

```suggestion
You can use variables to pass data between requests and tests, for example if you are [chaining requests](https://www.postman.com/postman/workspace/postman-team-collections/collection/1559645-81122f8b-5e07-4760-9504-f4387f45d2bc) in a collection.
```

File: src/pages/docs/sending-requests/variables.md [hunk 7]
```diff
@@ -59,121 +59,114 @@ Variables enable you to store and reuse values in your requests and scripts. By
...
-* Collection
-* Environment
-* Data
-* Local
+In order from broadest to narrowest, these scopes are: _global_, _collection_, _environment_, _data_, and _local_.
 
-<img alt="Variable Scope" src="https://assets.postman.com/postman-docs/var-scope.jpg" width="350px"/>
+Scope | Description | Notes
+--- | --- | ---
+**Global variables** | Enable you to access data between collections, requests, test scripts, and environments. Global variables are available throughout a [workspace](/docs/collaborating-in-postman/using-workspaces/creating-workspaces/). | Since global variables can create confusion, you should use them sparingly. For example, use a global variable to test something or when your project is at an early prototyping stage.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:110]: Can we reword the "confusion" sentence to make it more specific and convey why they can be confusing?

File: src/pages/docs/sending-requests/variables.md [hunk 8]
```diff
@@ -59,121 +59,114 @@ Variables enable you to store and reuse values in your requests and scripts. By
...
+**Collection variables** | Available throughout the requests in a collection and are independent of environments. Collection variables don't change based on the selected environment. | Collection variables are suitable if you are using a single environment, for example for auth or URL details.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:111]:
```suggestion
**Collection variables** | Available throughout the requests in a collection and are independent of environments. Collection variables don't change based on the selected environment. | Collection variables are suitable if you're using a single environment, for example for auth or URL details.
```

File: src/pages/docs/sending-requests/variables.md [hunk 9]
```diff
@@ -59,121 +59,114 @@ Variables enable you to store and reuse values in your requests and scripts. By
...
+**Environment variables** | Enable you to tailor your processing to different environments, for example local development versus testing or production. One environment can be active at a time. |  If you have a single environment, using collection variables can be more efficient, but environments enable you to specify [role-based access levels](/docs/sending-requests/managing-environments/#working-with-environments-as-a-team).
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:112]: I'd rephrase "Enable you to tailor your processing to different environments", "processing" reads a bit confusing to me

File: src/pages/docs/sending-requests/variables.md [hunk 10]
```diff
@@ -59,121 +59,114 @@ Variables enable you to store and reuse values in your requests and scripts. By
...
+**Data variables** | Come from external CSV and JSON files to define data sets you can use when running collections with [Newman](/docs/running-collections/using-newman-cli/) or the [Collection Runner](/docs/running-collections/intro-to-collection-runs/). | Data variables have current values, which don't persist beyond request or collection runs.
+**Local variables** | Temporary. These are accessed in your request scripts. Local variable values are scoped to a single request or collection run, and are no longer available when the run is complete. |  Local variables are suitable if you need a value to override all other variable scopes but don't want the value to persist once execution has ended.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:114]: This definition doesn't flow the same way as the others with the one word first sentence, might be good to reword a bit.

Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:114]: Maybe consider formatting this list like the brief one under #variable-types since it reads like sentences versus name|statement. 

> * **Data variables** come from external CSV...

File: src/pages/docs/sending-requests/variables.md [hunk 11]
```diff
@@ -59,121 +59,114 @@ Variables enable you to store and reuse values in your requests and scripts. By
...
+1. Select <img alt="Save icon" src="https://assets.postman.com/postman-docs/icon-save.jpg" width="16px" style="vertical-align:middle;margin-bottom:5px"> **Save** to confirm your changes.
 
-> You can also edit an environment by navigating to the workspace it resides in and selecting **Environments** from the left sidebar, then selecting your environment.
+<img alt="Environment editor" src="https://assets.postman.com/postman-docs/environment-editor-secret-v9.13.jpg">
 
-<!-- -->
+### Changing from secret to default variable type
 
-> You must have [editor](/docs/collaborating-in-postman/roles-and-permissions/) access on a workspace (for global variables) or environment (for environment variables) to control variable type.
+You must have [Editor](/docs/collaborating-in-postman/roles-and-permissions/) access on a workspace (for global variables) or environment (for environment variables) to control variable type. Editors can change the variable type from secret to default at any time, and vice versa. When you change a variable's type from secret back to default, you must confirm by selecting **Change type**.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:149]: Currently we don't consistently capitalize roles, might be good to discuss for style guide.

File: src/pages/docs/sending-requests/variables.md [hunk 12]
```diff
@@ -183,96 +176,119 @@ To create a variable at any scope from the request builder:
...
-pm.variables.set("variable_key", "variable_value");
-```
+> If you don't have Editor access to an environment, your script code will affect the current value but won't be synced or shared with your team.
+
+For instructions on how to use variables in pre-request or test scripts, see [Using variables in scripts](#using-variables-in-scripts).
 
-Local variables don't persist between sessions, but allow you to override all other scopes temporarily, during the execution of a request or collection / monitor run. For example, if you need to process a temporary test value for a single request or collection run locally, and don't want the value to sync with your team or remain available when the request / collection has finished running, you can use a local variable.
+### Specifying variable detail
+
+You can add and edit variables at any time. All you need to include for a new variable is a name. You can choose to supply an initial value, but you can also set it later, including from scripts. Use the checkbox to the left of a variable to activate or deactivate a variable.
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:285]:
```suggestion
You can add and edit variables at any time. All you need to include for a new variable is a name. You can choose to supply an initial value, but you can also set it later, including from [scripts](https://learning.postman.com/docs/writing-scripts/intro-to-scripts/). Use the checkbox to the left of a variable to activate or deactivate a variable.
```

File: src/pages/docs/sending-requests/variables.md [hunk 13]
```diff
@@ -183,96 +176,119 @@ To create a variable at any scope from the request builder:
...
+
+Initial values are shared when you share a collection or environment. Current values are local and not synced or shared. See [Sharing and persisting data](#sharing-and-persisting-data) for more on local versus synced variables.
 
-## Accessing variables
+## Using variables
 
-You can use double curly braces to reference variables throughout the Postman user interface. For example, to reference a variable named "username" in your request auth settings, you could use the following syntax with double curly braces around the name:
+You can use double curly braces to reference variables throughout Postman. For example, to reference a variable named "username" in your request auth settings, you would use the following syntax with double curly braces around the name:
```
Comment (claireriley) [inline src/pages/docs/sending-requests/variables.md:291]:
```suggestion
You can use double curly braces to reference variables throughout Postman. For example, to reference a variable named "username" in your request authorization settings, you would use the following syntax with double curly braces around the name:
```

**Review by claireriley — COMMENTED** [review]
Looks good so far! Added a few notes

File: src/pages/docs/sending-requests/variables.md (hunk 10 above)
Comment (HeatherStenson) [reply src/pages/docs/sending-requests/variables.md]: Good call, moving this back to a list.

**Review by malvikach — APPROVED** [review]

**Review by jkonrath-postman — APPROVED** [review]
LGTM!
